providers/implementations/exchange/dh_exch.c

```diff
@@ -392,7 +397,11 @@ static int dh_set_ctx_params(void *vpdhctx, const OSSL_PARAM params[])
         str = name;
         if (!OSSL_PARAM_get_utf8_string(p, &str, sizeof(name)))
             return 0;
-        pdhctx->kdf_cekalg = OPENSSL_strdup(name);
+        if (name != NULL) {
```

Comment:
This is wrong if. You'd have to do `if (p->data != NULL && p->data_size != 0) {` instead and do it before the OSSL_PARAM_get_utf8_string() call. Also I see there is missing unconditional `OPENSSL_free(pdhctx->kdf_cekalg); pdhctx->kdf_cekalg = NULL;` to clean up any previous data in the cekalg.